slsim/Sources/supernovae.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+import numpy.random as random
+import sncosmo
+from slsim.Util import param_util
+
+
+class Supernovae(sncosmo.Model):
+    """Class describing elliptical galaxies."""
+
+    def __init__(
+        self,
+        source,
+        redshift,
+        sn_type,
+        absolute_mag,
+        absolute_mag_band,
+        absolute_mag_zpsys,
+    ):
+        """
+
+        :param source: The model for the spectral evolution of the source. If a string
+            is given, it is used to retrieve a `~sncosmo.Source` from
+            the registry.
+        :type source: `~sncosmo.Source` or str
+        :param redshift: The redshift of the source.
+        :type redshift: float
+        :param sn_type: Supernova type (Ia, Ib, Ic, IIP, etc.)
+        :type sn_type: str
+        :param absolute_mag: Absolute magnitude of the supernova
+        :type absolute_mag: float
+        :param absolute_mag_band: Band used to normalize to absolute magnitude
+        :type absolute_mag_band: str or `~sncosmo.Bandpass`
+        :param absolute_mag_zpsys: AB or Vega
+        :type absolute_mag_zpsys: str
+        """
```

Comment:
Can you develop this class as a class which takes list of simulated supernovae (astropy table) and manages it as you want. You can write a code to simulate supernovae population outside of this class and within the Source directory. Does it make sense?

Reply:
Hi @jpierel14, I think you can develop Supernovae class to generate supernovae catalog in the form given in /slsim/Sources/SupernovaeData/supernovae_data.pkl. Then, we can call that class inside Lens class whenever supernovae catalog is needed. Thank you.

Reply:
@nkhadka21 sorry this conversation is a bit out of date now but is there still some sort of other interface you'd be interested in here?